CodeEdit/Features/QuickOpen/ViewModels/FuzzySearch.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  FuzzySearch.swift
+//  CodeEdit
+//
+//  Created by Tommy Ludwig on 04.05.23.
+//
+
+import Foundation
+
+enum FuzzySearch {
```

Comment:
Maybe add some general documentation to the enum? Which also brings the question to, why is this an `enum`?

Reply:
There's quite a lot of docs for each function in the enum, so I don't see why additional docs should be added.

If I'm guessing right, @activcoding used an enum here to namespace the functions. This is a practice sometimes done in Swift as enums can't be instantiated, which is useful for this use case. You can read about it [here](https://www.swiftbysundell.com/articles/powerful-ways-to-use-swift-enums/)

Reply:
The enum can have associated values to represent different search contexts. For example, you could define cases like `case fileSearch` or `case documentSearch`, each with their own associated values for performing search operations on different types of files or documents.
Because I planted to also use the fuzzy search for the Commands Overlay, which doesn’t sort an array of URLs I think.

Reply:
@Wouter01 Okay understandable.

@activcoding Why wouldn't you use polymorphism instead of enums in that case?

Reply:
That would be a valid approach as well, but to be honest I didn’t think about that decision to long. We can use one or the the other, the outcome should be the same. Depending what is best for you.

Reply:
True, but it's generally accepted, that switch cases are bad practice when OOP principles can be used. [Explaination](https://refactoring.guru/smells/switch-statements)

Reply:
> True, but it's generally accepted, that switch cases are bad practice when OOP principles can be used. [Explaination](https://refactoring.guru/smells/switch-statements)

In my opinion using polymorphism is a bit overkill here. If we'd go that route, I would prefer using POP instead of OOP, as Swift is more targeted towards the former.

Reply:
I think its a really small thing that can be changed with out an effort. I don't care what we use, I'm fine with both. Decide what you would rather work with and please don't make a mountain out of a molehill.

Reply:
> > True, but it's generally accepted, that switch cases are bad practice when OOP principles can be used. [Explaination](https://refactoring.guru/smells/switch-statements)
> 
> 
> 
> In my opinion using polymorphism is a bit overkill here. If we'd go that route, I would prefer using POP instead of OOP, as Swift is more targeted towards the former. 

I have to agree with Wouter here, it is probably a bit of overkill, and Swift is more POP than OOP. Since this also isn't really a issue with the PR and more of a enhancement to the PR. We should just approve it and maybe revisit later.

Reply:
Oh yeah for sure, it's not PR blocking imo, but I also want to look out for these kinds of things.